src/test/statistics/chunk_statistics/min_max_filter_test.cpp

```diff
@@ -0,0 +1,91 @@
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "base_test.hpp"
+#include "gtest/gtest.h"
+
+#include "utils/assert.hpp"
+
+#include "statistics/chunk_statistics/min_max_filter.hpp"
+#include "types.hpp"
+
+namespace opossum {
+
+template <typename T>
+class MinMaxFilterTest : public ::testing::Test {
+ protected:
+  void SetUp() override {
+    _values = pmr_vector<T>{-1000, 2, 3, 4, 7, 8, 10, 17, 123456};
+    _min_value = *std::min_element(std::begin(_values), std::end(_values));
+    _max_value = *std::max_element(std::begin(_values), std::end(_values));
+    _in_between = static_cast<T>(_min_value + 0.5 * (_max_value - _min_value));
+    _before_range = _min_value - 1;
+    _after_range = _max_value + 1;
+  }
+
+  pmr_vector<T> _values;
+  T _before_range, _min_value, _max_value, _after_range, _in_between;
+};
+
+template <>
+class MinMaxFilterTest<std::string> : public ::testing::Test {
+ protected:
+  void SetUp() override {
+    _values = pmr_vector<std::string>{"aa", "bb", "b", "bbbbba", "bbbbbb", "bbbbbc", "c"};
+    _min_value = *std::min_element(std::begin(_values), std::end(_values));
+    _max_value = *std::max_element(std::begin(_values), std::end(_values));
+    _in_between = "ba";
+    _before_range = "a";
+    _after_range = "cc";
+  }
+
+  pmr_vector<std::string> _values;
+  std::string _before_range, _min_value, _max_value, _after_range, _in_between;
+};
+
+using FilterTypes = ::testing::Types<int, float, double, std::string>;
+TYPED_TEST_CASE(MinMaxFilterTest, FilterTypes);
+
+TYPED_TEST(MinMaxFilterTest, CanPruneOnBounds) {
+  auto filter = std::make_unique<MinMaxFilter<TypeParam>>(this->_values.front(), this->_values.back());
+
+  for (const auto& value : this->_values) {
+    EXPECT_FALSE(filter->can_prune({value}, PredicateCondition::Equals));
+  }
+
+  EXPECT_TRUE(filter->can_prune({this->_before_range}, PredicateCondition::LessThan));
```

Comment:
A one-line comment in front of each of these blocks that tells us what it tests would be great.